src/content/configuration/other-options.md

```diff
@@ -446,3 +448,94 @@ module.exports = {
   }
 };
 ```
+
+## snapshot
+
+`object`
+
+Options affecting how file system snapshots are created and invalidated.
+
+__webpack.config.js__
+
+```javascript
+const path = require('path');
+module.exports = {
+  // ...
+  snapshot: {
+    managedPaths: [path.resolve(__dirname, '../node_modules')],
+    immutablePaths: [],
+    buildDependencies: {
+      hash: true,
+      timestamp: true
+    },
+    module: {
+      timestamp: true
+    },
+    resolve: {
+      timestamp: true
+    },
+    resolveBuildDependencies: {
+      hash: true,
+      timestamp: true
+    }
+  }
+};
+```
+
+### `managedPaths`
+
+`[string]`
+
+List of paths that are managed by a package manager and can be trusted to not be modified otherwise.
+
+### `immutablePaths`
+
+`[string]`
+
+List of paths that are managed by a package manager and contain version or hash in their paths so all files are immutable.
+
+### `buildDependencies`
+
+`object = {hash boolean = true, timestamp boolean = true}`
+
+Snapshots for build dependencies when using the persistent cache.
+
+- `hash`: Compare content hashes to determine invalidation (more expensive, but changes less often).
+- `timestamp`: Compare timestamps to determine invalidation.
+
+Both `hash` and `timestamp` are optional.
+
+- `{ hash: true }`: Good for CI caching with a fresh checkout which doesn't keep timestamps and hashes are used.
+- `{ timestamp: true }`: Good for local development caching.
+- `{ timestamp: true, hash: true }`: Good for both cases mentioned above, but has a small performance hit for the initial build since the timestamps are compared first then mismatch hashes are compared which allows cheap check before file need to be read for hash.
```

Comment:
It is. And I don't quite understand the last sentence.

Here's a new one I have with the help of your suggestion:

> Good for both cases mentioned above. Timestamps are compared first, which is cheap because webpack doesn't need to read files to compute their hashes. Content hashes will be compared only when timestamps are the same, which leads to a small performance hit for the initial build.

Any suggest?

Reply:
Yes reads good!